workflows/detaxizer.nf

```diff
@@ -105,95 +119,175 @@ workflow DETAXIZER {
     //
     // MODULE: Run fastp
     //
+    if (params.preprocessing) {
+
     FASTP (
         RENAME_FASTQ_HEADERS_PRE.out.fastq,
         [],
+        [],
         params.fastp_save_trimmed_fail,
         []
     )
+
+    ch_fastq_for_classification = FASTP.out.reads
     ch_versions = ch_versions.mix(FASTP.out.versions.first())
+    } else {
+        ch_fastq_for_classification = RENAME_FASTQ_HEADERS_PRE.out.fastq
+    }
+    //////////////////////////////////////////////////
+    //  Classification
+    //////////////////////////////////////////////////
 
-    //
-    // MODULE: Prepare Kraken2 Database
-    //
-    ch_kraken2_db = Channel.fromPath(params.kraken2db).map {
-            item -> [['id': "kraken2_db"], item]
-        }
-    KRAKEN2PREPARATION (
-        ch_kraken2_db
-    )
-    ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions)
+    if ( params.classification_kraken2_post_filtering || (!params.classification_kraken2 && !params.classification_bbduk) || (params.classification_kraken2) ){
+        //
+        // MODULE: Prepare Kraken2 Database
+        //
+        ch_kraken2_db = Channel.fromPath(params.kraken2db).map {
+                item -> [['id': "kraken2_db"], item]
+            }
 
-    //
-    // MODULE: Run Kraken2
-    //
+        KRAKEN2PREPARATION (
+            ch_kraken2_db
+        )
+        ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions.first())
+    }
 
-    KRAKEN2_KRAKEN2 (
-        FASTP.out.reads,
-        KRAKEN2PREPARATION.out.db.first(),
-        params.save_output_fastqs,
-        true
-    )
-    ch_versions = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first())
 
-    //
-    // MODULE: Parse the taxonomy from the kraken2 report and return all subclasses of the tax2filter
-    //
-    PARSE_KRAKEN2REPORT(
-        KRAKEN2_KRAKEN2.out.report.take(1)
-    )
-    ch_versions = ch_versions.mix(PARSE_KRAKEN2REPORT.out.versions)
+    if ((!params.classification_bbduk && !params.classification_kraken2) || (params.classification_kraken2)) {
 
-    //
-    // MODULE: Isolate the hits for a certain taxa and subclasses
-    //
-    ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+        //
+        // MODULE: Run Kraken2
+        //
+        KRAKEN2_KRAKEN2 (
+            ch_fastq_for_classification,
+            KRAKEN2PREPARATION.out.db.first(),
+            params.save_output_fastqs,
+            true
+        )
+        ch_versions = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first())
```

Comment:
Again no use of MultiQC?

Reply:
I'll add `Usage of MultiQC` as an issue, tag `enhancement`.